spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java

```diff
@@ -225,28 +238,45 @@ public void doWith(Method method) throws IllegalArgumentException,
 		}
 		Method mostSpecificMethod = AopUtils
 				.getMostSpecificMethod(method, invocation.getThis().getClass());
-		NewSpan annotation = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod);
-		if (annotation == null) {
+		NewSpan newSpan = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod, NewSpan.class);
+		ContinueSpan continueSpan = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod, ContinueSpan.class);
+		if (newSpan == null && continueSpan == null) {
 			return invocation.proceed();
 		}
-		Span span = null;
-		boolean hasLog = StringUtils.hasText(annotation.log());
+		Span span = tracer().getCurrentSpan();
+		String log = log(newSpan, continueSpan);
+		boolean hasLog = StringUtils.hasText(log);
 		try {
-			span = spanCreator().createSpan(invocation, annotation);
+			if (newSpan != null) {
+				span = spanCreator().createSpan(invocation, newSpan);
+			}
 			if (hasLog) {
-				span.logEvent(annotation.log() + ".start");
+				spanTagAnnotationHandler().addAnnotatedParameters(invocation);
+				span.logEvent(log + ".start");
 			}
 			return invocation.proceed();
 		} finally {
 			if (span != null) {
 				if (hasLog) {
-					span.logEvent(annotation.log() + ".end");
+					span.logEvent(log + ".end");
```

Comment:
make sense, so in case of using annotation it should be like 
```
@ContinueSpan(log = "ingredientsAggregation")
public void startIngredientsAggregation() {
}
```
and we can took just method name by default, right?